specs/commandline/sign.md

```diff
@@ -73,20 +75,24 @@ Notation uses empty JSON object `{}` as the default configuration content, and t
 # - A signing plugin is installed. See plugin documentation (https://github.com/notaryproject/notaryproject/blob/main/specs/plugin-extensibility.md) for more details.
 # - Configure the signing plugin as instructed by plugin vendor.
 
-# Add a default signing key referencing the remote key identifier, and the plugin associated with it.
-notation key add --default --name <key_name> --plugin <plugin_name> --id <remote_key_id>
-
-# sign an artifact stored in a registry
-notation sign <registry>/<repository>@<digest>
+notation sign --plugin <plugin_name> --id <remote_key_id> <registry>/<repository>@<digest>
```

Comment:
By looking at this design, basically we are going to support two paths:
1. User adds a new key into `signingkeys.json` with `notation key add` -> Sign using `notation sign --key <key_name>`
2. User does NOT add a new key into `signingkeys.json`, instead Sign directly with `notation sign --plugin <plugin_name> --id <remote_key_id>`

Based on above, I have a couple of questions:
1. What is the expected behavior of this command: `notation sign --plugin <plugin_name> --id <remote_key_id> --key <key_name>`, where the key with <key_name> might have a different <plugin_name> and/or <remote_key_id> than the one specified by user. Looks like `[--plugin, --id]` needs to be mutual exclusive with `--key`?
2. In the 1st path, when adding a key using `notation key add` command, we can do `notation key add --plugin-config`, later on sign with `notation sign --plugin-config`. Under the hood, Notation would merge these two plugin-config maps, and configs from `notation sign` would override the ones from `notation key add`. (Related code: https://github.com/notaryproject/notation-go/blob/8c3ed9217dc8025983f64a5510249a94dc009a86/signer/plugin.go#L170) 
Looks like in the 2nd path, we are asking users to do it themselves by combining all the necessary plugin configs (key + sign) into `notation sign` command's `--plugin-config`. Do we need to mention it somewhere in our command line help page?

Reply:
>  1. What is the expected behavior of this command: `notation sign --plugin <plugin_name> --id <remote_key_id> --key <key_name>`, where the key with <key_name> might have a different <plugin_name> and/or <remote_key_id> than the one specified by user. Looks like `[--plugin, --id]` needs to be mutual exclusive with `--key`?

Fail the sign request of both `--key` is used wither with `--plugin` or `--id`


>  2. In the 1st path, when adding a key using `notation key add` command, we can do `notation key add --plugin-config`, later on sign with `notation sign --plugin-config`. Under the hood, Notation would merge these two plugin-config maps, and configs from `notation sign` would override the ones from `notation key add`. (Related code: https://github.com/notaryproject/notation-go/blob/8c3ed9217dc8025983f64a5510249a94dc009a86/signer/plugin.go#L170)
>   When a user is not using `--key` and using `--plugin` + `--id` flag(s) for signing, it means that the user doesn't want a lookup in `sinigngkeys.json` file. Also, technically notation cannot perform a lookup in `sinigngkeys.json`  because there is no `--key` for lookup. This means that we don't have the problem of merging `--plugin-config`. To summarize we will only use plugin-config passed in the command flag